pkg/cmd/pr/checkout/checkout.go

```diff
@@ -258,3 +289,40 @@ func executeCmds(client *git.Client, cmdQueue [][]string) error {
 	}
 	return nil
 }
+
+func selectPR(httpClient *http.Client, baseRepo ghrepo.Interface, prompter shared.Prompter, cs *iostreams.ColorScheme) (*api.PullRequest, error) {
+	listResult, err := list.ListPullRequests(httpClient, baseRepo, shared.FilterOptions{Entity: "pr", State: "open", Fields: []string{
+		"number",
+		"title",
+		"state",
+		"url",
+		"headRefName",
+		"headRepositoryOwner",
+		"isCrossRepository",
+		"isDraft",
+		"createdAt",
+	}}, 30)
```

Comment:
From personal experience, our team is quite small, and we tidy up our PR queue quickly so I rarely need more than the first 30 PRs. 

But let me know how we want to design this feature, to cover your use cases also :)